src/pieces/messageCount.ts

```diff
@@ -1,23 +1,79 @@
-import { Client, TextChannel } from 'discord.js';
+import { Client, TextChannel, Role, Message, MessageEmbed } from 'discord.js';
 import { generateLogEmbed } from '@lib/utils';
 import { DatabaseError } from '@lib/types/errors';
-import { LOG, PREFIX, DB } from '@root/config';
+import { LOG, PREFIX, DB, ROLES } from '@root/config';
+import { SageUser } from '@lib/types/SageUser';
+
+const xpRatio = 1.25;
+const startingColor = 80;
+const greenIncrement = 8;
+const maxGreen = '00ff00';
+const maxLevel = 20;
 
 async function register(bot: Client): Promise<void> {
 	const errLog = await bot.channels.fetch(LOG.ERROR) as TextChannel;
-	bot.on('message', msg => {
+	bot.on('message', async msg => {
```

Comment:
It may be worth making this callback into its own function so you can catch errors similar to how the serverlog is setup.